navigator_engine/common/graph_visualizer.py

```diff
@@ -0,0 +1,209 @@
+def create_visualizer(flask_app, dash_app) -> None:
+
+    dash_app.layout = html.Div(children=[
+        html.Div(
+            id='dropdown-container',
+            children=[
+                dcc.Dropdown(
+                    id='graph-dropdown',
+                    options=[],
+                    placeholder='Select a graph',
+                )
+            ]
+        ),
+        html.Hr(),
+        html.Div(
+            id='cytoscape-container',
+            children=[
+                cyto.Cytoscape(
+                    id='cytoscape-figure',
+                    layout={'name': 'preset'},
+                    style={'width': '100%', 'height': '400px'},
+                    elements=[
+                    ])
+            ]
+        ),
+        html.Hr(),
+        html.P(
+            id='tapNodeData-output',
+            children=['Click a node to see more information about it']
+        ),
+    ])
+
+    @dash_app.callback(
+        Output(component_id='tapNodeData-output', component_property='children'),
+        Input(component_id='cytoscape-figure', component_property='tapNodeData'))
+    def display_tap_node_data(data: dict) -> list:
+        if data:
+            return [data['infobox']]
+        else:
+            return []
+
+    @dash_app.callback(
+        Output(component_id='graph-dropdown', component_property='options'),
+        Input(component_id='graph-dropdown', component_property='placeholder'))
+    def load_dropdown_options(placeholder) -> list:
+        with flask_app.app_context():
+            graphs = model.Graph.query.all()
+
+        dropdown_options = []
+        for graph in graphs:
+            dropdown_options.append({'label': f'Graph {graph.id} | {graph.title}', 'value': graph.id})
+
+        return dropdown_options
+
+    @dash_app.callback(
+        Output(component_id='cytoscape-container', component_property='children'),
+        Input(component_id='graph-dropdown', component_property='value'))
+    def update_figure(graph_id: int) -> list:
+
+        if not graph_id:
+            return []
+
+        with flask_app.app_context():
+            graph = model.load_graph(graph_id=graph_id)
+            graph_x = graph.to_networkx()
+
+        root_nodes = [n for n, d in graph_x.in_degree() if d == 0]
+        assert len(root_nodes) == 1
+
+        elements = []
+        for n in graph_x.nodes:
+            node = model.load_node(node_id=n.id)
+            if node.action and node.action.complete:
+                node_element = {'data': {'id': str(node.id),
+                                         'label': f'A {node.action.id} COMPLETE',
+                                         'infobox': f'Action {node.action.id} | '
+                                                    f'Node {node.id} | '
+                                                    f'{node.action.title}'
+                                         },
+                                'classes': 'red triangle'
+                                }
```

Comment:
Can you format this dictionary and all dictionaries below like this please: 
```suggestion
                node_element = {
                    'classes': 'red triangle',
                    'data': {
                        'id': str(node.id),
                        'label': f'A {node.action.id} COMPLETE',
                        'infobox': f'Action {node.action.id} | '
                                   f'Node {node.id} | '
                                   f'{node.action.title}'
                    }
                 }
```